build/bazelutil/bazel-generate.sh

```diff
@@ -2,14 +2,60 @@
 
 set -euo pipefail
 
+# files_unchanged_from_upstream takes file globs as arguments and checks
+# whether all these files are unchanged from the upstream master branch.
+# It is best effort.
+# This optimization can be disabled by setting env var
+# COCKROACH_BAZEL_FORCE_GENERATE=1.
+files_unchanged_from_upstream () {
+  if [ "${COCKROACH_BAZEL_FORCE_GENERATE:-}" = 1 ]; then
+    return 1
+  fi
+  
+  if ! which git >/dev/null; then
+    return 1
+  fi
+
+  # First, figure out the correct remote.
+  UPSTREAM=$(git remote -v | grep 'github.com[/:]cockroachdb/cockroach.*(fetch)' | awk '{print $1}') || return 1
+  if [ -z "$UPSTREAM" ]; then
+    return 1
+  fi
+
+  # Find the upstream commit which this branch is based on.
+  BASE=$(git merge-base $UPSTREAM/master HEAD 2>/dev/null) || return 1
+  if [ -z "$BASE" ]; then
+    return 1
+  fi
+
+  # Check if the files are unchanged.
+  DIFF=$(git diff --no-ext-diff --name-only $BASE "$@") || return 1
```

Comment:
Nit: I think if you add a `--` before `"$@"`, that should make it clear to `git` that those arguments are supposed to be paths.